### ISSUE TITLE
Multiple videos not playing correctly

### ISSUE THREAD

**Issue by wrngwrld** [media-kit/media-kit#964]
Whenever I try to run the example [09.seamless.dart](https://github.com/media-kit/media-kit/blob/main/media_kit_test/lib/tests/09.seamless.dart) and set the AudioTrack to auto instead of none. I always get this stutter on page changes/the videos are not playing. Is there a fix for this? It probably is a problem of the dispose method, as disabling the dispose in this example does make it run smoother.

https://github.com/user-attachments/assets/1dc79314-26b7-42a4-91f5-028a7e1b6747

**Comment by alexmercerind** [comment]
Yeah... this is a thing with iOS. I haven't figured out how to allow multiple audio sources to run in parallel.

Be assured that this is not stutter... but iOS system pauses another audio as soon as one one starts playing.

**Comment by wrngwrld** [comment]
I'm actually not too sure, that it is iOS pausing the audio rather than something weird happening in the dispose when the audio is enabled. When I removed these lines (https://github.com/media-kit/media-kit/blob/264a54f23e6e0b0c564800de39f547b074f1b28a/media_kit_test/lib/tests/09.seamless.dart#L89-L103), the players no longer dispose, and the lag between them disappears as well.

**Comment by alexmercerind** [comment]
With audio enabled?

**Comment by wrngwrld** [comment]
Yes, with the audio enabled

**Comment by alexmercerind** [comment]
Interesting...

**Comment by alexmercerind** [comment]
Let us know if you find something... I don't own an iOS, try to comment out things inside dispose method maybe.

**Comment by wrngwrld** [comment]
The weird thing is that the only thing, that fixes it is by commenting out this line https://github.com/media-kit/media-kit/blob/264a54f23e6e0b0c564800de39f547b074f1b28a/media_kit_test/lib/tests/09.seamless.dart#L93

So there probably is some code in the platform dispose method, that makes it lag. I'll try to take a look into that code when I have time :)

**Comment by alexmercerind** [comment]
I don't see the same on any other platform... I still doubt its an audio issue. We do modify --aid etc. in dispose

**Comment by alexmercerind** [comment]
Try building against main branch now.

**Comment by wrngwrld** [comment]
Still no improvement :(

**Comment by doersoul** [comment]
Calling player.dispose(), will trigger the suspension of the audio session, causing other players to also pause playing

solution

```dart
  /// Disposes the [Player] instance & releases the resources.
  @override
  Future<void> dispose({bool synchronized = true}) {
    Future<void> function() async {
      if (disposed) {
        throw AssertionError('[Player] has been disposed');
      }

      await waitForPlayerInitialization;
      await waitForVideoControllerInitializationIfAttached;

      await NativeReferenceHolder.instance.remove(ctx);

      // tips: modified here
      if (Platform.isIOS) {
        await _command(['quit']);
      } else {
        await stop(notify: false, synchronized: false);
      }

      disposed = true;

      await super.dispose();

      Initializer(mpv).dispose(ctx);

      // tips: modified here
      if (!Platform.isIOS) {
        Future.delayed(const Duration(seconds: 5), () {
          mpv.mpv_terminate_destroy(ctx);
        });
      }
    }

    if (synchronized) {
      return lock.synchronized(function);
    } else {
      return function();
    }
  }

```
